### DISCUSSION TITLE
ironbar in diffrent layers ?

### DISCUSSION THREAD

**Discussion by leoVici9** [JakeStanger/ironbar#579] [Q&A] [answered]
hi , 
 new ironbar user here on hyprland and until now really happy. however when trying the auotohide functionality  i noticed that when revealing a hidden bar , it causes the hyprland windows to resize because the bar takes screen layer real estate. is there a possibility to have the autohiding bars in the overlay layer ?
 
 use cases :  
1) use  the launcher module in a bottom bar as a dock with autohide. currently not possible with autohide turned on because of ripple effects on windows.
2) use cairo module in a side bar with auto hide (or sys_info module) but again a bar in the top layer claims reaal estate. a bar in the overlay layer would be very conveniet way to display this info when needed. however windows will be resized which looks ugly and thus not really usable that way.

regards
Bert

**Comment by JakeStanger** [comment]
Hey, currently this is not possible but I can definitely see the use-case for this and it should be a quick addition. I'll look into making the following changes:

- Add a config option to set each bar's layer
- If possible, default the bar to the overlay layer if it starts hidden
- For bonus points, I might be able to expose an IPC/CLI command to change a bar's layer at runtime (no idea how GTK or compositors will feel about that one yet)

I will open a feature request to track.

**Reply by JakeStanger** [reply]
#582

**Comment by leoVici9** [comment, marked as answer]
you must be dying for some feed back :-) 

Yes it works great , the only strange thing is having to set exclusive_zone to false for items in layer background or start_hidden to true for items in layer overlay.  this could be more implicit. but for now very happy , see attached image
![screenshot_15062024_165533-s](https://github.com/JakeStanger/ironbar/assets/160249523/973684af-47a5-4248-aac1-3937528c5d01) 

now i need a tranparant background on the clock :-)

Bert

**Reply by JakeStanger** [reply]
Thanks for sharing. That's the first time I've seen Ironbar be used for anything other than a bar, and cool to see my old clock widget revived! I'd love to see a show-and-tell post when you're done.

Your point about setting `exclusive_zone` based on the layer makes sense. That'll be an easy change to feel free to open an issue.

**Reply by leoVici9** [reply]
yeah seeing the other discussions about a bar being a bar , it surprised me that you agreed to add the back ground as a layer :-) but yes ironbar is now a real conky replacement. still working on porting a sysinfo conky ( or use sysinfo module in popup bar) and doing some theming but then i can share in show and tell :-)

bert

**Reply by JakeStanger** [reply]
Awesome, I look forward to it!

It's a balance between maintainability. Letting users set the layer was trivial to add in, and doesn't really change the scope of supported features or the project, so I was happy to add it in.
